src/components/chips/js/chipsController.js

```diff
@@ -0,0 +1,196 @@
+     * @type {boolean}
+     */
+    this.useMdChipAppend = false;
+
+    /**
+     * Whether the Chip buffer is driven by an input element provided by the
+     * caller.
+     * @type {boolean}
+     */
+    this.hasInputElement = false;
+  }
+
+
+  /**
+   * Handles the keydown event on the input element: <enter> appends the
+   * buffer to the chip list, while backspace removes the last chip in the list
+   * if the current buffer is empty.
+   * @param event
+   */
+  MdChipsCtrl.prototype.defaultInputKeydown = function(event) {
+    switch (event.keyCode) {
+      case this.$mdConstant.KEY_CODE.ENTER:
+        event.preventDefault();
+        this.appendChipBuffer();
+        break;
+      case this.$mdConstant.KEY_CODE.BACKSPACE: // backspace
+        if (!this.chipBuffer) {
+					event.preventDefault();
+          // TODO(typotter): Probably want to open the previous one for edit instead.
+          this.removeChip(this.items.length - 1);
+        }
+        break;
+      default:
+    }
+  };
+
+
+  /**
+   * Sets the selected chip index to -1.
+   */
+  MdChipsCtrl.prototype.resetSelectedChip = function() {
+    this.selectedChip = -1;
+  };
+
+
+  /**
+   * Append the contents of the buffer to the chip list. This method will first
+   * call out to the md-chip-append method, if provided
+   */
+  MdChipsCtrl.prototype.appendChipBuffer = function() {
+    var newChip = this.getChipBuffer();
+    if (this.useMdChipAppend && this.mdChipAppend) {
+      newChip = this.mdChipAppend({'$chip': newChip});
+    }
+    this.items.push(newChip);
+    this.resetChipBuffer();
+  };
+
+
+  /**
+   * Sets whether to use the md-chip-append expression. This expression is
+   * bound to scope and controller in {@code MdChipsDirective} as
+   * {@code mdChipAppend}. Due to the nature of directive scope bindings, the
+   * controller cannot know on its own/from the scope whether an expression was
+   * actually provided.
+   */
+  MdChipsCtrl.prototype.useMdChipAppendExpression = function() {
+    this.useMdChipAppend = true;
+  };
+
+
+  /**
+   * Gets the input buffer. The input buffer can be the model bound to the
+   * default input item {@code this.chipBuffer}, the {@code selectedItem}
+   * model of an {@code md-autocomplete}, or, through some magic, the model
+   * bound to any inpput or text area element found within a
+   * {@code md-input-container} element.
+   * @return {Object|string}
+   */
+  MdChipsCtrl.prototype.getChipBuffer = function() {
+    if (this.mdAutocompleteCtrl) {
+      this.$log.error('md-autocomplete not yet supported');
+    } else if (this.hasInputElement) {
+      this.$log.error('user-provided inputs not yet supported');
+    } else {
+      return this.chipBuffer;
+    }
+  };
+
+
+  /**
+   * Resets the input buffer.
+   */
+  MdChipsCtrl.prototype.resetChipBuffer = function() {
+    if (this.mdAutocompleteCtrl) {
+      this.$log.error('md-autocomplete not yet supported');
+    } else {
+      this.chipBuffer = '';
```

Comment:
Since Line 74 checks this `if (!this.chipBuffer)`, I do not recommend using

```js
this.chipBuffer = '';
```

instead use 

```js
this.chipBuffer = null;
```